src/Widgets/LocationPopover.vala

```diff
@@ -0,0 +1,248 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public class Tasks.LocationPopover : Tasks.EntryPopover<Tasks.Location?> {
+    private GtkChamplain.Embed map_embed;
+    private Gtk.SearchEntry search_entry;
+    private GLib.Cancellable search_cancellable;
+    private Granite.Widgets.ModeButton location_mode;
+    private Marker point;
+
+
+    public LocationPopover () {
+        Object (
+            icon_name: "location-inactive-symbolic",
+            placeholder: _("Set Location")
+        );
+    }
+
+    construct {
+        map_embed = new GtkChamplain.Embed () {
+            height_request = 120,
+            width_request = 220
+        };
+
+        point = new Marker () {
+            draggable = true
+        };
+
+        var marker_layer = new Champlain.MarkerLayer.full (Champlain.SelectionMode.SINGLE);
+        marker_layer.add_marker (point);
+
+        var map_view = map_embed.champlain_view;
+        map_view.zoom_level = 10;
+        map_view.goto_animation_duration = 500;
+        map_view.add_layer (marker_layer);
+        map_view.center_on (point.latitude, point.longitude);
+
+        var map_frame = new Gtk.Frame (null);
+        map_frame.add (map_embed);
+
+        location_mode = new Granite.Widgets.ModeButton ();
+        location_mode.append_text (_("Arrival"));
+        location_mode.append_text (_("Departure"));
+
+        search_entry = new Gtk.SearchEntry () {
+            placeholder_text = _("John Smith OR Example St."),
+            hexpand = true
+        };
+
+        var grid = new Gtk.Grid () {
+            margin = 6,
+            row_spacing = 6,
+            column_spacing = 6
```

Comment:
As in Dialogs, margins in popovers should be 12px, and since there are no columns in this layout we don't need to set column spacing:
```suggestion
            margin = 12
            row_spacing = 12
```